e2e/_suites/fleet/fleet.go

```diff
@@ -1259,24 +1268,38 @@ func deployAgentToFleet(installer ElasticAgentInstaller, containerName string, t
 			"service": service,
 			"tag":     serviceTag,
 		}).Error("Could not run the target box")
-		return err
+		return nil, err
 	}
 
 	err = installer.PreInstallFn()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	err = installer.InstallFn(containerName, token)
+	var fleetConfig *FleetConfig
```

Comment:
Didn't actually tried, but I think it would be a bit simpler to have just 1 kind of config and 1 way of creating it, eg: `func NewAgentConfig(token, fleetServerMode bool) (*AgentConfig, error)` 

Credentials, url and port can also be hardcoded in the only place they are used.

Reply:
👍

Reply:
Implemented in https://github.com/elastic/e2e-testing/pull/900/commits/f5a9f46f8153aaf9595a4821f1a50951c4452a39